File: Data Structures & Algorithms/eating-bananas/submission-8.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        def checkBanana(val):
            time = 0
            i = 0
            while time<=h and i<len(piles):
                time += math.ceil(piles[i]/val)
                i+=1
            return time<=h
        l,r = 1,max(piles)
        res = None
        while l<=r:
            m = (l+r)//2
            if checkBanana(m):
                res = m
                r =m-1
            else:
                l =m+1
        return res

                
        

File: Data Structures & Algorithms/insert-new-interval/submission-8.java
class Solution {
    public int[][] insert(int[][] intervals, int[] newInterval) {

        List<int[]> res = new ArrayList<>();
        for(int i= 0;i<intervals.length;i++){
            int[] interval = intervals[i];
            int startNew= newInterval[0], endNew = newInterval[1], endPrev = interval[1],
            startPrev = interval[0];
            if(startNew>endPrev){
                res.add(interval);
            }  else if(endNew<startPrev) {
                res.add(newInterval);
                for(int j=i;j<intervals.length;j++){
                    res.add(intervals[j]);
                }
                return res.toArray(new int[0][]);

            } else if(startNew<=endPrev){
                newInterval = new int[]{Math.min(startPrev,startNew),Math.max(endNew,endPrev)};
            }
        }
        res.add(newInterval);
        return res.toArray(new int[0][]);

        
    }

    // if startNew>endPrev
    //     insert prev
    // elif startNew<=endPrev{
    //     update newInterval
    // } else if endNew<startPrev {
    //     insert new then merge rest of list and return

    // } 

}
